src/lib/misc.c

```diff
@@ -1139,4 +1140,76 @@ rnp_filename(const char *path)
     } else {
         return res + 1;
     }
-}
\ No newline at end of file
+}
+
+/* find the time - in a specific %Y-%m-%d format - using a regexp */
+int
+grabdate(const char *s, int64_t *t)
+{
+    static regex_t r;
+    static int     compiled;
+    regmatch_t     matches[10];
+    struct tm      tm;
+
+    if (!compiled) {
+        compiled = 1;
+        (void) regcomp(
+          &r, "([0-9][0-9][0-9][0-9])[-/]([0-9][0-9])[-/]([0-9][0-9])", REG_EXTENDED);
+    }
+    if (regexec(&r, s, 10, matches, 0) == 0) {
+        (void) memset(&tm, 0x0, sizeof(tm));
+        tm.tm_year = (int) strtol(&s[(int) matches[1].rm_so], NULL, 10);
+        tm.tm_mon = (int) strtol(&s[(int) matches[2].rm_so], NULL, 10) - 1;
+        tm.tm_mday = (int) strtol(&s[(int) matches[3].rm_so], NULL, 10);
+        *t = mktime(&tm);
+        return 1;
+    }
+    return 0;
+}
+
+/* get expiration in seconds */
+uint64_t
+get_duration(const char *s)
```

Comment:
Don't know. As stated before - these 3 functions were just moved to the more suitable place.